src/ir/local-graph.h

```diff
@@ -121,8 +121,17 @@ struct LocalGraph {
   Function* func;
   std::set<Index> SSAIndexes;
 
-  // A map of each get to the sets relevant to it.
-  GetSetsMap getSetsMap;
+  // A map of each get to the sets relevant to it. This is mutable so that
+  // getSets() can be const.
+  mutable GetSetsMap getSetsMap;
+
+  // The internal implementation of the flow analysis used to compute
+  // getSetsMap.
+  struct LocalGraphFlower;
+  // This could be a unique_ptr, but the forward declaration is not compatible
+  // with that. It could alternatively be a shared_ptr, but that runs into what
+  // seems to be a false positive of clang's (but not gcc's) UBSan.
```

Comment:
This doesn't seem right. We use a unique pointer to an incomplete implementation type just fine in `TypeBuilder`.

Reply:
Well, this was very mysterious, but eventually I found a StackOverflow comment that said "try making an empty destructor and moving it out of the header". And... that worked.

Reply:
Yay, gotta love C++.

Reply:
I do understand this in the end, I guess. If the destructor is in a file where we only have a forward declaration, then the compiler can't emit the destructor - it doesn't know what to call. So this makes sense, but the compiler errors in both gcc and clang did not help me here.